Demos/Device/ClassDriver/CCID/Descriptors.c

```diff
@@ -0,0 +1,219 @@
+			.TotalInterfaces          = 1,
+
+			.ConfigurationNumber      = 1,
+			.ConfigurationStrIndex    = NO_DESCRIPTOR,
+
+			.ConfigAttributes         = (USB_CONFIG_ATTR_RESERVED | USB_CONFIG_ATTR_SELFPOWERED),
+
+			.MaxPowerConsumption      = USB_CONFIG_POWER_MA(100)
+		},
+		.CCID_Interface =
+		{
+			.Header                 = {.Size = sizeof(USB_Descriptor_Interface_t), .Type = DTYPE_Interface},
+
+			.InterfaceNumber        = INTERFACE_ID_CCID,
+			.AlternateSetting       = 0x00,
+
+			.TotalEndpoints         = 2,
+
+			.Class                  = CCID_CSCP_CCIDClass, 
+			.SubClass               = CCID_CSCP_NoSpecificSubclass,
+			.Protocol               = CCID_CSCP_NoSpecificProtocol,
+
+			.InterfaceStrIndex      = NO_DESCRIPTOR
+		},
+		.CCID_SmartCard =
+		{
+			.Header                 = {.Size = sizeof(USB_CCID_Descriptor_t), .Type = DTYPE_FunctionalDescriptor},
+			.CCID  					= CCID_CURRENT_SPEC_RELEASE_NUMBER,
+			.MaxSlotIndex			= 0x00,
+			.VoltageSupport			= CCID_VOLTAGESUPPORT_5V,
+			.Protocols				= CCID_PROTOCOLS_T1,
+			.DefaultClock			= 16000, //16MHz
+			.MaximumClock			= 16000, //16MHz
+			.NumClockSupported		= 0,
+			.DataRate				= 307200,
+			.MaxDataRate			= 307200,
+			.NumDataRatesSupported	= 0, 
+			.MaxIFSD				= 2038,
+			.SynchProtocols			= 0,
+			.Mechanical				= 0,
+			.Features				= 0x0400fe,
+			.MaxCCIDMessageLength	= 0x0c00,
+			.ClassGetResponse		= 0xff,
+			.ClassEnvelope			= 0xff,
+			.LcdLayout				= 0,
+			.PINSupport				= 0,
+			.MaxCCIDBusySlots		= 1
+			
+		},
+		.CCID_BulkInEndpoint =
+		{
+			.Header                 = {.Size = sizeof(USB_Descriptor_Endpoint_t), .Type = DTYPE_Endpoint},
+
+			.EndpointAddress        = CCID_IN_EPADDR,
+			.Attributes             = EP_TYPE_BULK,
+			.EndpointSize           = CCID_EPSIZE,
+			.PollingIntervalMS      = 0x05
+		},
+
+		.CCID_BulkOutEndpoint =
+		{
+			.Header                 = {.Size = sizeof(USB_Descriptor_Endpoint_t), .Type = DTYPE_Endpoint},
+
+			.EndpointAddress        = CCID_OUT_EPADDR , 
+			.Attributes             = EP_TYPE_BULK,
+			.EndpointSize           = CCID_EPSIZE,
+			.PollingIntervalMS      = 0x05
+		}
+};
+
+/** Language descriptor structure. This descriptor, located in FLASH memory, is returned when the host requests
+ *  the string descriptor with index 0 (the first index). It is actually an array of 16-bit integers, which indicate
+ *  via the language ID table available at USB.org what languages the device supports for its string descriptors.
+ */
+const USB_Descriptor_String_t PROGMEM LanguageString = USB_STRING_DESCRIPTOR_ARRAY(LANGUAGE_ID_ENG);
+
+/** Manufacturer descriptor string. This is a Unicode string containing the manufacturer's details in human readable
+ *  form, and is read out upon request by the host when the appropriate string ID is requested, listed in the Device
+ *  Descriptor.
+ */
+const USB_Descriptor_String_t PROGMEM ManufacturerString = USB_STRING_DESCRIPTOR(L"CCID Manufacturer");
```

Comment:
I think I'll change all the existing demo manufacturer descriptors to "LUFA Library" rather than my own name, can you also change that here please?